pkg/scheduler/worker/workerpool.go

```diff
@@ -173,6 +187,10 @@ func (w *Pool) invokeProcessableOps(workerPool *ants.PoolWithFunc) (int, error)
 		idx++
 	}
 	w.logger.Infof("Worker pool assigned %d of %d processable operations to workers", idx, opsCnt)
+	err = w.occupancyRepo.UpdateWorkerPoolOccupancy(w.poolID, workerPool.Running())
```

Comment:
I would highly recommend to move the whole tracking of the worker pool occupany into a dedicated component and not to add it on top of the worker pools. The occupancy tracking could happen in intervals and also the synchronisation of worker pools with running mothership-pods has to happen in intervals.